### PR TITLE
chore: update eslint + prettier rules

### LINKED ISSUE #253: Add a more opiniated linter: Prettier
In some external PRs, I see some code formatting differences than from the rest of the codebase. It's of course not the fault of the contributor, but it's due to the current linting rules being a bit too lenient.

The idea of this issue is to remove semistandard, and enforce `@typescript-eslint/recommended` + Prettier.

The config I would like:

### .eslintrc.js

```
module.exports = {
  extends: [
    'plugin:react/recommended',
    'plugin:@typescript-eslint/recommended',
    'prettier/@typescript-eslint',
    'plugin:prettier/recommended'
  ],
  parser: '@typescript-eslint/parser',
  plugins: ['react', '@typescript-eslint', 'prettier'],
  settings: {
    react: {
      version: "detect"
    }
  },
};
```

### .prettierrc.js

```
module.exports = {
  semi: true,
  singleQuote: true,
  tabWidth: 2,
};
```

Note: you might need to `yarn add eslint-plugin-react eslint-plugin-prettier prettier` beforehand

Note 2: (Bonus) If you have some ideas of other rules we could add on top of this, please propose here too. I'm thiking those that will: sort imports alphabetically, sort props alphabetically, sort object deconstruction alphabetically.

### PR DESCRIPTION
**Pull request by ftonato**
#### Summary:

- Add a more opiniated linter: Prettier
-----
Closes #253

### CODE REVIEW

**Comment by amaurymartiny** [comment]
Thanks! I hope it was clear in the issue, but the scope of #253 also contains fixing the ~500 linting issues we have now.

**Comment by ftonato** [comment]
Hey there,
I saw it, but I prefer to separate things and I'll send another Pull-request with these fix, hold for that... ⏳

**Comment by amaurymartiny** [comment]
Ah, I see! I would still prefer 1 PR for all this, because I will not merge a PR with a red CI.

**Comment by ftonato** [comment]
Ahh okay okay, just wait for it (please). I'll fix here  =)

**Comment by ftonato** [comment]
Hello @amaurymartiny,

I have a few doubts about this linting issues, let me know what do you think about them:

Some of the most common **errors**:

1. (33 occurrences) Forbidden non-null assertion _@typescript-eslint/no-non-null-assertion_
2. (02 occurrences) Type assertion on object literals is forbidden, use a type annotation instead _@typescript-eslint/no-object-literal-type-assertion_

Some of the most common **warnings**:

3. (173 occurrences) Missing return type on function _@typescript-eslint/explicit-function-return-type_
> I think we can ignore this for a while, disabling the rule for all files, right?

4. (33 occurrences) Unexpected any. Specify a different type _@typescript-eslint/no-explicit-any_

----

In the file [App/Screens/Search/fetchAlgolia.ts#47](https://github.com/amaurymartiny/shoot-i-smoke/blob/master/App/Screens/Search/fetchAlgolia.ts#L47), we have the following problem:
What is the impact of changing this to a camel case?

**Comment by amaurymartiny** [comment]
> (33 occurrences) Forbidden non-null assertion @typescript-eslint/no-non-null-assertion

This one is important! It might catch some bugs, so I wish to keep it.

Basically, for example https://github.com/amaurymartiny/shoot-i-smoke/blob/9641cf6437f9043fd95295a67f8ebcb77687fdc0/App/Screens/Home/Home.tsx#L44

you would need to write:

```typescript
if (!api) {
  throw new Error('Home.tsx only reachable when `api` is defined.'); // Or some message like this
}

const cigarettesPerDay = api.shootISmoke.cigarettes; 
```

> (02 occurrences) Type assertion on object literals is forbidden, use a type annotation instead @typescript-eslint/no-object-literal-type-assertion

I would like to keep these two too. I didn't find where they were, if you need help for those two I can have a look too.

> Warnings

This PR seems bigger than I thought. In this case you can do

```
  rules: {
    '@typescript-eslint/no-explicit-any': 0,
    '@typescript-eslint/explicit-function-return-type': 0
  },
```

for now in .eslintrc.js, we can do those in a separate PR.

File: App/Screens/About/About.tsx [hunk 1]
```diff
@@ -27,12 +27,11 @@ import {
 import { ScrollIntoView, wrapScrollView } from 'react-native-scroll-into-view';
 import { scale } from 'react-native-size-matters';
 import { NavigationInjectedProps } from 'react-navigation';
-
-import { Box } from './Box';
 import { BackButton } from '../../components';
 import { i18n } from '../../localization';
 import { trackScreen } from '../../util/amplitude';
 import * as theme from '../../util/theme';
+import { Box } from './Box';
```
Comment (amaurymartiny) [inline App/Screens/About/About.tsx]: I like this rule! putting deeper imports first. Is it some semi-official rule, or your custom?

Comment (ftonato) [reply App/Screens/About/About.tsx]: I like too, but is part of my custom settings :)

**Comment by ftonato** [comment]
I found these errors:

- App/Screens/Screens.tsx
  44:10  error  Type assertion on object literals is forbidden, use a type annotation instead  @typescript-eslint/no-object-literal-type-assertion

- App/stores/fetchGpsPosition/fetchGpsPosition.ts
  48:10  error  Type assertion on object literals is forbidden, use a type annotation instead  @typescript-eslint/no-object-literal-type-assertion

Do you have any suggestion or would you like to fix them?

File: App/Screens/Home/Footer/Footer.tsx [hunk 2]
```diff
@@ -63,7 +80,7 @@ export function Footer (props: FooterProps) {
       Share.share({
         title: i18n.t('home_share_title'),
         message: i18n.t('home_share_message', {
-          cigarettes: api!.shootISmoke.cigarettes.toFixed(2)
+          cigarettes: (api && api.shootISmoke.cigarettes.toFixed(2)) || 0
```
Comment (amaurymartiny) [inline App/Screens/Home/Footer/Footer.tsx]:
```suggestion
          cigarettes: api ? api.shootISmoke.cigarettes.toFixed(2) : 0
```

I prefer the `?:` syntax, it reads better imo. Could you replace all the other instances too?

File: package.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
     "haversine": "^1.1.1",
     "i18n-js": "^3.3.0",
     "io-ts": "^2.0.1",
+    "prettier": "^1.18.2",
```
Comment (amaurymartiny) [inline package.json]: prettier should go in dev deps

File: package.json [hunk 4]
```diff
@@ -63,7 +64,7 @@
     "eslint-plugin-node": "^10.0.0",
```
Comment (amaurymartiny) [inline package.json]: If I'm not mistaken, we can remove:
- eslint-plugin-standard
- eslint-plugin-promise
- eslint-plugin-node
- eslint-plugin-import
- eslint-config-standard
- eslint-config-semistandard

File: package.json (hunk 3 above)
Comment (ftonato) [reply package.json]: my mistake =(

**Comment by ftonato** [comment]
Time to merge my friend :unicorn:

**Comment by amaurymartiny** [comment]
Thanks, almost there but not yet! There are 2 errors on Travis, surely because of the master merge. The `DistanceUnit` type comes from `stores/distanceUnit.tsx` now

**Comment by ftonato** [comment]
My bad, I did merge using GitHub instead of the command-line interface, but I hope that it's correct now :rocket: :microscope:

File: App/Screens/Home/Header/Header.tsx [hunk 5]
```diff
@@ -86,3 +63,56 @@ const styles = StyleSheet.create({
...
+          api={api}
+          currentLocation={currentLocation}
+          numberOfLines={2}
+        />
+        <View style={styles.distance}>
+          {isTooFar && <Image source={alert} style={styles.warning} />}
+          <Text style={styles.distanceText}>
+            {i18n.t('home_header_air_quality_station_distance', {
+              distanceToStation: distance,
+              distanceUnit
```
Comment (amaurymartiny) [inline App/Screens/Home/Header/Header.tsx]:
```suggestion
              distanceUnit: shortDistanceUnit
```

merging error

File: App/Screens/Details/Distance/Distance.tsx [hunk 6]
```diff
@@ -16,19 +16,27 @@
...
+    justifyContent: 'center'
+  },
+  distance: {
+    ...theme.title,
+    color: 'white'
+  }
+});
+
+export function Distance(props: DistanceProps) {
+  const distanceUnit = i18n.t('distance_unit');
```
Comment (amaurymartiny) [inline App/Screens/Details/Distance/Distance.tsx]: Merge error. You need to keep

```
const { localizedDistanceUnit } = useDistanceUnit();
const distanceUnit = localizedDistanceUnit('short');
```

File: App/Screens/Home/Header/Header.tsx [hunk 7]
```diff
@@ -86,3 +63,56 @@ const styles = StyleSheet.create({
     marginTop: scale(-2) // FIXME We shouldn't need that, with `alignItems: 'center'` on .distance
   }
 });
+
+export function Header(props: HeaderProps) {
+  const { api } = useContext(ApiContext);
+  const { currentLocation, isGps } = useContext(CurrentLocationContext);
+  const { onChangeLocationClick } = props;
+
+  const distanceUnit = i18n.t('distance_unit');
```
Comment (amaurymartiny) [inline App/Screens/Home/Header/Header.tsx]: Merge error. You need to keep `const shortDistanceUnit = localizedDistanceUnit('short');`

File: App/Screens/About/About.tsx [hunk 8]
```diff
@@ -142,18 +204,6 @@ export function About (props: AboutProps) {
         </Text>
       </View>
 
-      <View style={styles.distance}>
```
Comment (amaurymartiny) [inline App/Screens/About/About.tsx]: Ah, sorry, forgot about this one. Another merge error: This big chunk has been forgotten.

File: App/Screens/ErrorScreen/ErrorScreen.tsx [hunk 9]
```diff
@@ -14,32 +14,48 @@
...
+  }
+});
 
-export function ErrorScreen (props: ErrorScreenProps) {
+export function ErrorScreen(props: ErrorScreenProps) {
   const { error } = useContext(ErrorContext);
 
   trackScreen('ERROR');
 
-  useEffect(() => {
```
Comment (amaurymartiny) [inline App/Screens/ErrorScreen/ErrorScreen.tsx]: This chunk too

File: App/Screens/ShareScreen/ShareScreen.tsx [hunk 10]
```diff
@@ -38,24 +64,15 @@ export function ShareScreen (props: ShareScreenProps) {
...
         });
 
-        if (Platform.OS === 'ios') {
-          await Share.share({
-            url: uri
-          });
-        } else {
-          await Sharing.shareAsync(uri, {
-            mimeType: 'image/png'
-          });
```
Comment (amaurymartiny) [inline App/Screens/ShareScreen/ShareScreen.tsx]: This chunk has been forgotten

File: App/Screens/Home/Footer/Footer.tsx [hunk 11]
```diff
@@ -15,48 +15,75 @@
...
     track('HOME_SCREEN_DETAILS_CLICK');
     navigation.navigate('Details');
   }
 
-  function handleShare () {
+  function handleShare() {
     track('HOME_SCREEN_SHARE_CLICK');
 
-    props.navigation.navigate('ShareModal');
+    // Share doesn't currently support images on Android, so the text version
```
Comment (amaurymartiny) [inline App/Screens/Home/Footer/Footer.tsx]: Merge error

File: App/Screens/Details/Details.tsx [hunk 12]
```diff
@@ -19,29 +19,28 @@ import { StyleSheet, View } from 'react-native';
...
 let stationMarker: Marker | undefined;
 
-export function Details (props: DetailsProps) {
+export function Details(props: DetailsProps) {
   const { navigation } = props;
 
   const [showMap, setShowMap] = useState(false);
   const { api } = useContext(ApiContext);
-  const { currentLocation: _currentLocation } = useContext(CurrentLocationContext);
-  const { distanceUnit } = useDistanceUnit();
```
Comment (amaurymartiny) [inline App/Screens/Details/Details.tsx]: This got dropped too

**Comment by ftonato** [comment]
I will never merge again using GitHub 🐛

**Pull request merged**